### ISSUE TITLE
While creating a alb listener rule for the http-request-method through moto, I am getting ""An error occurred (ValidationError) when calling the CreateRule operation: Condition field 'http-request-method' must be one of '[path-pattern, host-header]" Exception.

### ISSUE THREAD

**Issue by satishgampala** [spulec/moto#2544]
While creating a alb listener rule for the http-request-method through moto, 

I am getting "An error occurred (ValidationError) when calling the CreateRule operation: Condition field 'http-request-method' must be one of '[path-pattern, host-header]" Exception.

Sample code snippet:

```
import boto3
from moto import mock_elbv2

alb_client = boto3.client('elbv2', region_name = 'eu-central-1')


@mock_elbv2
def create_alb_listener_rule():
    alb_client.create_rule(
        Priority=10,
        ListenerArn='string',
        Conditions=[
            {
                "Field": "HttpRequestMethodConfig",
                "HttpRequestMethodConfig": {
                    "Values": [
                        "POST"
                    ]
                }
            }
        ],
        Actions=[
            {
                'Type': 'forward',
                'TargetGroupArn': "XXXXXXXXXXXXXXXXXXX"
            }
        ]
    )
```

Exception:

```
"An error occurred (ValidationError) when calling the CreateRule operation: Condition field 'http-request-method' must be one of '[path-pattern, host-header]"
```

Is moto comaptible with the latest boto3 versions???

**Comment by satishgampala** [comment]
Even after trying with "Field": "http-request-method", then also it fails

**Comment by spulec** [comment]
Thanks for opening. Setting this as a feature request for us to implement `http-request-method`

**Comment by tabramczyk** [comment]
I have same problem with `"Field": "http-header"`.

There are more options now:
> `The field in the HTTP request. The following are the possible values:
>
> - http-header
> - http-request-method
> - host-header
> - path-pattern
> - query-string
> - source-ip`

Source:
https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/elbv2.html#ElasticLoadBalancingv2.Client.create_rule

**Comment by bblommers** [comment]
This is now fixed in #4066
